core/tests/snapshots.rs

```diff
@@ -610,4 +625,244 @@ mod tests {
+            None,
+        )?;
+        let incremental_snapshot_package = snapshot_utils::process_accounts_package_pre(
+            incremental_snapshot_package,
+            Some(bank.get_thread_pool()),
+            Some(incremental_snapshot_base_slot),
+        );
+        snapshot_utils::archive_snapshot_package(
+            &incremental_snapshot_package,
+            snapshot_config.maximum_snapshots_to_retain,
+        )
+    }
+
+    fn restore_from_incremental_snapshot_and_check_banks_are_equal(
+        bank: &Bank,
+        last_full_snapshot_slot: Slot,
+        snapshot_config: &SnapshotConfig,
+        accounts_dir: PathBuf,
+        genesis_config: &GenesisConfig,
+    ) -> snapshot_utils::Result<()> {
+        let (
+            full_snapshot_archive_slot,
+            (incremental_snapshot_archive_base_slot, incremental_snapshot_archive_slot),
+            deserialized_bank,
+        ) = restore_from_incremental_snapshot(snapshot_config, accounts_dir, genesis_config)?;
+
+        assert_eq!(
+            full_snapshot_archive_slot,
+            incremental_snapshot_archive_base_slot
+        );
+        assert_eq!(full_snapshot_archive_slot, last_full_snapshot_slot);
+        assert_eq!(incremental_snapshot_archive_slot, bank.slot(),);
+        assert_eq!(*bank, deserialized_bank);
+
+        Ok(())
+    }
+
+    fn restore_from_incremental_snapshot(
+        snapshot_config: &SnapshotConfig,
+        accounts_dir: PathBuf,
+        genesis_config: &GenesisConfig,
+    ) -> snapshot_utils::Result<(Slot, (Slot, Slot), Bank)> {
+        let full_snapshot_archive_info = snapshot_utils::get_highest_full_snapshot_archive_info(
+            &snapshot_config.snapshot_package_output_path,
+        )
+        .ok_or_else(|| Error::new(ErrorKind::Other, "no full snapshot"))?;
+
+        let incremental_snapshot_archive_info =
+            snapshot_utils::get_highest_incremental_snapshot_archive_info(
+                &snapshot_config.snapshot_package_output_path,
+                *full_snapshot_archive_info.slot(),
+            )
+            .ok_or_else(|| Error::new(ErrorKind::Other, "no incremental snapshot"))?;
+
+        info!("Restoring bank from full snapshot slot: {}, and incremental snapshot slot: {} (with base slot: {})",
+        full_snapshot_archive_info.slot(), incremental_snapshot_archive_info.slot(), incremental_snapshot_archive_info.base_slot());
+
+        let (deserialized_bank, _) = snapshot_utils::bank_from_snapshot_archives(
+            &[accounts_dir],
+            &[],
+            &snapshot_config.snapshot_path,
+            full_snapshot_archive_info.path(),
+            Some(incremental_snapshot_archive_info.path()),
+            snapshot_config.archive_format,
+            genesis_config,
+            None,
+            None,
+            AccountSecondaryIndexes::default(),
+            false,
+            None,
+            accounts_db::AccountShrinkThreshold::default(),
+            false,
```

Comment:
This flag to not check the hash was introduced here: https://github.com/solana-labs/solana/pull/17939/files#diff-b5dead0a38e5f63447595bbeda1428d3aa851d33954343bb1bc38b4b00add5fdR149, not sure why it's not safe to check the hash here. 

Maybe the hash is verified elsewhere already? Would be good to check the accounts state is being verified to match the snapshot hash somewhere in the test.

Reply:
heh, looks like in the normal path, the hash is verified *after*  the call to `bank_from_snapshot_archives()` https://github.com/solana-labs/solana/blob/86bda02fc4116660d779654971b1a83471be0bea/ledger/src/bank_forks_utils.rs#L149-L171

Reply:
@carllin Sure thing. Can I break this out into a separate PR/issue instead of fixing inside this PR?

Reply:
If extracting the above logic into a function and calling it from the test works, I think it makes sense to include in this PR, as testing the accounts hash seems important to the snapshot tests.

If somethings broken/hash doesn't verify, then a separate PR makes sense to me.

Reply:
Carl and I talked. I've created a new issue to track this request: #18973.